wcfsetup/install/files/lib/acp/form/ReactionTypeEditForm.class.php

```diff
@@ -36,4 +38,31 @@ public function readParameters() {
 			}
 		}
 	}
+	
+	/**
+	 * Sets the form data based on the current form object.
+	 */
+	protected function setFormObjectData() {
+		parent::setFormObjectData(); 
+		
+		if (empty($_POST)) {
+			$this->uploadFormField->value([
+				new UploadFile(WCF_DIR . 'images/reaction/' . $this->formObject->iconFile, basename($this->formObject->iconFile), true, true, true)
```

Comment:
Do you think it might be possible that `UploadFormField::value()` could do this itself so that we wouldn't have to bother doing it ourselves when using this form field?

Reply:
Theoretically we could introduce a method in the DBO object, which assembles the UploadFiles for us. However, this would probably have to be implemented there individually for most objects, since the files are stored differently.